papis/commands/doctor.py

```diff
@@ -633,6 +638,64 @@ def biblatex_required_keys_check(doc: papis.document.Document) -> List[Error]:
             if not any(key in doc or aliases.get(key) in doc for key in keys)]
 
 
+BIBLATEX_EXPECTED_KEYS_IGNORED = {
+    "papis_id", "author_list", "editor_list", "citations", "doc_url",
+    "files", "ref", "time-added", "type", "tags"
+} | frozenset(papis.config.getlist("bibtex-ignore-keys"))
+BIBLATEX_EXPECTED_KEYS_CHECK_NAME = "biblatex-expected-keys"
+
+
+def biblatex_expected_keys_check(doc: papis.document.Document) -> List[Error]:
+    """
+    Check that all keys in info.yaml are expected by biblatex processors.
+
+    In most circumstances, using arbitrary keys is completely inconsequential.
+    This check simply helps users keep strict control over metadata.
+
+    :returns: an error per document key that is not a known key
+    """
+    from papis.bibtex import bibtex_keys
+    folder = doc.get_main_folder() or ""
+
+    errors = bibtex_type_check(doc)
+    if errors:
+        return [e._replace(name=BIBLATEX_REQUIRED_KEYS_CHECK_NAME) for e in errors]
+
+    def make_fixer(key: str) -> FixFn:
+        def fixer() -> None:
+            config_file = papis.config.get_config_file()
+            config = papis.config.get_configuration()
+            lib = papis.config.get_lib_name()
+            logger.info("[FIX] Adding '%s' to %s.bibtex-ignore-keys", key, lib)
+
+            if os.path.exists(config_file):
+                logger.warning("file contents (whitespace, comments) may change")
+                if not papis.tui.utils.confirm("Proceed with overwrite?"):
+                    raise Exception("fix canceled by user")
+
+            ignores = papis.config.getlist("bibtex-ignore-keys")
+            ignores.append(key)
+            config[lib]["bibtex-ignore-keys"] = str(ignores)
+
+            with open(config_file, "w") as configfile:
+                config.write(configfile)
+            logger.info("Configuration file saved at '%s'.", config_file)
```

Comment:
This seems very convoluted and unsafe for an automated fix. I would recommend just extending the error message with a suggestion e.g. `add '{key}' to 'bibtex-ignore-keys' in your configuration file to silence this error`?

Reply:
One thing we could do here, is extend the `--edit` functionality to allow opening the configuration file instead if the check requests it?